webapp/testing.md

```diff
@@ -1,8 +1,38 @@
 # Component Testing
 
-We are using `Jest` as our test runner, along with `vue-test-utils`.
+## Linting
 
-Head over and check out the documentation on [Jest](https://jestjs.io/docs/en/getting-started.html)
+We use [ESLint](https://eslint.org/) to make sure all developers follow certain code guidelines when writing JavaScript.
 
-Also, check out [vue-test-utils](https://vue-test-utils.vuejs.org/)
+Most code editors offer an ESLint plugin which helps detect mistakes already while you are writing code. To run the linter manually before pushing up new code type `yarn lint` into your terminal. Most minor issues can be fixed automatically with the command `yarn lint --fix`.
 
+## Unit tests
+
+We write unit tests with the help of [Jest](https://jestjs.io/) and [Vue Test Utils](https://vue-test-utils.vuejs.org/) to make sure our components work in the way they should. In these tests we usually check that a certain input leads to the expected output. They are used to test _functionality_.
+
+To run all tests use the command `yarn test` in the `/webapp` directory. Other useful commands are:
+- `yarn test -t test-name` to run tests including `test-name` in their file or test names
+- `yarn test -o` to run tests related to files that have been changed since the latest commit
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-31T10:57:11Z" title="Thursday, October 31st 2019, 11:57:11 am +01:00">Oct 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,8 +1,37 @@
 # Component Testing
 
-We are using `Jest` as our test runner, along with `vue-test-utils`.
+## Linting
 
-Head over and check out the documentation on [Jest](https://jestjs.io/docs/en/getting-started.html)
+We use [ESLint](https://eslint.org/) to make sure all developers follow certain code guidelines when writing JavaScript.
 
-Also, check out [vue-test-utils](https://vue-test-utils.vuejs.org/)
+Most code editors offer an ESLint plugin which helps detect mistakes already while you are writing code. To run the linter manually before pushing up new code type `yarn lint` into your terminal. Most minor issues can be fixed automatically with the command `yarn lint --fix`.
 
+## Unit tests
+
+We write unit tests with the help of [Jest](https://jestjs.io/) and [Vue Test Utils](https://vue-test-utils.vuejs.org/) to make sure our components work in the way they should. In these tests we usually check that a certain input leads to the expected output. They are used to test _functionality_.
+
+To run all tests use the command `yarn test` in the `/webapp` directory. Other useful commands are:
+- `yarn test -t test-name` to run tests including `test-name` in their file or test names
+- `yarn test -o` to run tests related to files that have been changed since the latest commit
```


Nice!

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-31T15:53:23Z" title="Thursday, October 31st 2019, 4:53:23 pm +01:00">Oct 31, 2019</time>_

---


```suggestion
- `yarn test -o` to run tests related to files that have been changed since the latest commit
- `yarn run path/to/component.spec.js` to run a single test file
```